src/parse.rs

```diff
@@ -2139,6 +2162,17 @@ where
         }
     }
 
+    // Expect one of multiple possible tokens, advances the token stream
+    fn expect_one_of(&mut self, toks: Vec<Tok>) -> Result<(Tok, (usize, usize)), ParseError> {
```

Comment:
Let's have this function take a slice rather than a vector to avoid taking ownership and cloning the vector here outside of the error path.